client/src/lib.rs

```diff
@@ -0,0 +1,336 @@
+    wallet::Wallet,
+};
+use linera_core::node::{ValidatorNode as _, ValidatorNodeProvider as _};
+use linera_views::store::WithError;
+use serde::ser::Serialize as _;
+use wasm_bindgen::prelude::*;
+use web_sys::{js_sys, wasm_bindgen};
+
+// TODO(12): convert to IndexedDbStore once we refactor Context
+type WebStorage =
+    linera_storage::DbStorage<linera_views::memory::MemoryStore, linera_storage::WallClock>;
+
+async fn get_storage() -> Result<WebStorage, <linera_views::memory::MemoryStore as WithError>::Error>
+{
+    linera_storage::DbStorage::initialize(
+        linera_views::memory::MemoryStoreConfig::new(1),
+        "linera",
+        b"",
+        Some(linera_execution::WasmRuntime::Wasmer),
+    )
+    .await
+}
+
+type PersistentWallet = linera_client::persistent::Memory<Wallet>;
+type ClientContext = linera_client::client_context::ClientContext<WebStorage, PersistentWallet>;
+type ChainClient =
+    linera_core::client::ChainClient<linera_rpc::node_provider::NodeProvider, WebStorage>;
+
+// TODO(13): get from user
+pub const OPTIONS: ClientOptions = ClientOptions {
+    send_timeout: std::time::Duration::from_millis(4000),
+    recv_timeout: std::time::Duration::from_millis(4000),
+    max_pending_message_bundles: 10,
+    wasm_runtime: Some(linera_execution::WasmRuntime::Wasmer),
+    max_concurrent_queries: None,
+    max_loaded_chains: nonzero_lit::usize!(40),
+    max_stream_queries: 10,
+    cache_size: 1000,
+    retry_delay: std::time::Duration::from_millis(1000),
+    max_retries: 10,
+    wait_for_outgoing_messages: false,
+    blanket_message_policy: linera_core::client::BlanketMessagePolicy::Accept,
+    restrict_chain_ids_to: None,
+    long_lived_services: false,
+
+    // TODO(linera-protocol#2944): separate these out from the
+    // `ClientOptions` struct, since they apply only to the CLI/native
+    // client
+    tokio_threads: Some(1),
+    command: linera_client::client_options::ClientCommand::Keygen,
+    wallet_state_path: None,
+    storage_config: None,
+    with_wallet: None,
+};
+
+#[wasm_bindgen(js_name = Wallet)]
+pub struct JsWallet(PersistentWallet);
+
+#[wasm_bindgen(js_class = "Wallet")]
+impl JsWallet {
+    /// Creates and persists a new wallet from the given JSON string.
+    ///
+    /// # Errors
+    /// If the wallet deserialization fails.
+    #[wasm_bindgen]
+    pub async fn create(wallet: &str) -> Result<JsWallet, JsError> {
+        Ok(JsWallet(PersistentWallet::new(serde_json::from_str(
+            wallet,
+        )?)))
+    }
+
+    /// Attempts to read the wallet from persistent storage.
+    ///
+    /// # Errors
+    /// If storage is inaccessible.
+    #[wasm_bindgen]
+    pub async fn read() -> Result<Option<JsWallet>, JsError> {
+        Ok(None)
+    }
+}
+
+/// The full client API, exposed to the wallet implementation.  Calls
+/// to this API can be trusted to have originated from the user's
+/// request.  This struct is the backend for the extension itself
```

Comment:
Nit:

```suggestion
/// request. This struct is the backend for the extension itself
```

(Also elsewhere.)